rcl_lifecycle/src/com_interface.c

```diff
@@ -108,6 +108,9 @@ rcl_lifecycle_com_interface_publisher_init(
 
   // initialize publisher
   rcl_publisher_options_t publisher_options = rcl_publisher_get_default_options();
+  publisher_options.qos.reliability = RMW_QOS_POLICY_RELIABILITY_RELIABLE;
+  publisher_options.qos.durability = RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL;
+  publisher_options.qos.depth = 1;
```

Comment:
```suggestion
  // transition event topic needs to be latched for the subscription joins later.
  publisher_options.qos.durability = RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL;
  publisher_options.qos.depth = 1;
```

changing publisher's QoS to transient local should not break any downstream subscriptions, https://docs.ros.org/en/rolling/Concepts/Intermediate/About-Quality-of-Service-Settings.html#id6.

IMO this transition event topic needs to be latched to deliver the latest state for the subscribers that join later. probably adding comments why we use transient local with depth 1 would be useful.

besides, [Managed Node Design](https://design.ros2.org/articles/node_lifecycle.html) also says the following.

> A topic should be provided to broadcast the new life cycle state when it changes. This topic must be latched.